packages/graphql-transformer-core/src/__tests__/util/amplifyUtils.test.ts

```diff
@@ -0,0 +1,31 @@
+import { getSanityCheckRulesFactory, SanityCheckRules } from '../../util/amplifyUtils';
+import { AmplifyCLIFeatureFlagAdapter } from '../../../../amplify-provider-awscloudformation/src/utils/amplify-cli-feature-flag-adapter';
+import { FeatureFlags } from 'amplify-cli-core';
+
+jest.mock('amplify-cli-core');
+
+describe('get sanity check rules', () => {
+  test('empty list when api is in create status', () => {
+    const ff_mock = new AmplifyCLIFeatureFlagAdapter();
+    (<any>FeatureFlags.getBoolean).mockReturnValue(true);
+    const sanityCheckRules: SanityCheckRules = getSanityCheckRulesFactory(true, ff_mock);
+    expect(sanityCheckRules.diffRules.length).toBe(0);
+    expect(sanityCheckRules.projectRules.length).toBe(0);
+  });
+
+  test('sanitycheck rule list when api is in update status and ff enabled', () => {
+    const ff_mock = new AmplifyCLIFeatureFlagAdapter();
+    (<any>FeatureFlags.getBoolean).mockReturnValue(true);
+    const sanityCheckRules: SanityCheckRules = getSanityCheckRulesFactory(false, ff_mock);
+    expect(sanityCheckRules.diffRules.length).toBe(4);
+    expect(sanityCheckRules.projectRules.length).toBe(1);
```

Comment:
Can we verify the rules themselves by name instead of the size

Reply:
I tried this with snapshot, but they were prinitng [functions] instead of Names.
Is there a way to print their names?